src/server/rpc/procedures/roboscape/robot.js

```diff
@@ -379,39 +381,139 @@ Robot.prototype.onMessage = function (message) {
     } else if (command === 'R' && message.length === 13) {
         this.sendToClient('range', {
             range: message.readInt16LE(11),
-        }, ['time', 'range']);
+        });
     } else if (command === 'T' && message.length === 19) {
         this.sendToClient('ticks', {
             left: message.readInt32LE(11),
             right: message.readInt32LE(15),
-        }, ['time', 'left', 'right']);
+        });
     } else if (command === 'D' && message.length === 15) {
         this.sendToClient('drive', {
             left: message.readInt16LE(11),
             right: message.readInt16LE(13),
-        }, ['time', 'left', 'right']);
+        });
     } else if (command === 'L' && message.length === 13) {
         this.sendToClient('set led', {
             led: message.readUInt8(11),
             command: message.readUInt8(12)
-        }, ['time', 'led', 'command']);
+        });
     } else if (command === 'F' && message.length === 12) {
         state = message.readUInt8(11);
         this.sendToClient('infra event', {
             left: (state & 0x2) == 0,
             right: (state & 0x1) == 0
-        }, ['time', 'left', 'right']);
+        });
     } else if (command === 'G' && message.length === 14) {
         this.sendToClient('infra light', {
             msec: message.readInt16LE(11),
             pwr: Math.round(100 - message.readUInt8(13) / 2.55)
-        }, ['time', 'msec', 'pwr']);
+        });
     } else {
         this._logger.log('unknown ' + this.ip4_addr + ':' + this.ip4_port +
             ' ' + message.toString('hex'));
     }
 };
 
+// handle user commands to the robot (through the 'send' rpc)
+Robot.prototype.onCommand = function(command) {
+    const cases = [
+        {
+            regex: /^is alive$/,
+            handler: () => {
+                this.sendToClient('alive', {});
+                return this.isAlive();
+            }
+        },
+        {
+            regex: /^beep (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.beep(+RegExp.$1, +RegExp.$2);
```

Comment:
these could be turned into one line arrow functions but keeping it this way to not change the intial behavior.